### PR TITLE
Added support for resetting an instance of the stack

### PR DESCRIPTION
**Pull request by marius-serban**
Submitting this PR for review, I know it's not ready yet because unit tests still need to be added.

See #49

### CODE REVIEW

**Comment by jessesquires** [comment]
Thanks @marius-serban! :smile: 

One major goal of this project is to keep it focused and simple, and keep classes small. So, I'm not 100% sold on this idea.

1. Does this need to be part of the library? What part of the current API prevents one from doing this already? The `storeCoordinator` is public, so you should be able to simply add/remove a store. And then there's `removeExistingModelStore()` that we discussed in #49.

2. This functionality should be implemented in a new class/struct/top-level function — not added to `CoreDataStack`. This makes the stack too smart. In fact, `CoreDataStack` doesn't even really know how to configure/setup itself — the `CoreDataStackFactory` does this. The stack should only be responsible for managing its properties (model, context, coordinator) and keeping contexts in sync.

**Comment by jessesquires** [comment]
Oops. Didn't mean to close. :grin:

**Comment by marius-serban** [comment]
I agree with you about simplicity, that's one of the things I like about this library. I'm going to move this out of `CoreDataStack` into a top level function and make use of `removeExistingModelStore()`. Then we'll see how it looks, it may be a simple clean addition.

File: JSQCoreDataKit/JSQCoreDataKit/CoreDataExtensions.swift [hunk 1]
```diff
@@ -142,3 +142,46 @@ public func deleteObjects <T: NSManagedObject>(objects: [T], inContext context:
...
+    let storeCoordinator = stack.storeCoordinator
+    let model = stack.model
+
+    stack.mainContext.reset()
+    stack.backgroundContext.reset()
+
+    var caughtError: NSError?
+    storeCoordinator.performBlockAndWait {
+        do {
+            if let store = storeCoordinator.persistentStores.first {
```
Comment (jessesquires) [inline JSQCoreDataKit/JSQCoreDataKit/CoreDataExtensions.swift]: let's do this check above at L158 and return early

**Comment by jessesquires** [comment]
Thanks @marius-serban ! Much simpler :smile: 

A few more notes:

- Not sure about `dispatch_group`. Almost positive we don't need this with the `performBlock:` APIs.
- Need to add unit tests for this. Let's make a new `ResetStackTests.swift` and follow the patterns in the existing tests

**Comment by jessesquires** [comment]
Oh -- an if you don't mind, let's squash your commits into 1 before merging :smile:

**Comment by marius-serban** [comment]
✅ done

**Comment by jessesquires** [comment]
@marius-serban this is awesome! :smile: thanks! will merge soon.

img

**Comment by jessesquires** [comment]
After merging, I'll release v2.2.0

**Comment by marius-serban** [comment]
Yay, can't wait!
ship it

**Pull request merged**